knative/src/duck/v1/knative_reference.rs

```diff
@@ -42,8 +47,20 @@ impl Into<ObjectReference> for KReference {
 }
 
 impl KReference {
-    pub fn resolve_uri(&self, _client: kube::Client) -> Result<url::Url, Error> {
-        let _object_reference: ObjectReference = self.clone().into();
+    pub async fn resolve_uri(
+        &self,
+        client: kube::Client,
+    ) -> Result<url::Url, Box<dyn std::error::Error>> {
```

Comment:
It would be nice to keep our error model in-line [with kube's](https://docs.rs/kube-client/0.74.0/src/kube_client/error.rs.html#9) by sticking with `thiserror` wrappers of Error.

If we add a Variant to [knative error](https://github.com/rusty-jules/knative-rs/blob/6c0ad0ac0c55ce465c53ce1e12958bf4dd701e4e/knative/src/error.rs#L4) with a `#[from]` `kube-client::Error` then it will automatically implement `TryFrom` which gives us back `?` operator without resorting to `Box<dyn std::error::Error>`. We may need to drop `Clone` and `Copy` derives, but that's ok!

Our error can look like this:
```rust
use kube::error::Error as KubeError;

#[derive(Error, Debug)]
pub enum Error {
    /// Discovery errors
    #[error("Error from discovery: {0}")]
    Discovery(#[source] DiscoveryError),
    /// Kube errors
    #[error("Error from kube: {0}")]
    KubeError(#[from] KubeError),
}
```

Reply:
Also, the `Discovery` Variant is admittedly not very thought out. Feel free to scratch or rework it when we add the actual Errors that can occur during `KReference` uri resolution!